src/main/java/Duke.java

```diff
@@ -1,10 +1,210 @@
+                System.out.println("Please indicate a valid task number! :)\n");
+            }
+        }
+
+        /**
+         * Handles the adding of the 3 different kinds of tasks.
+         * @param command Input command word
+         * @param contents Input task name
+         */
+        private static void addNewTask(CommandWord command, String contents){
+            Task newTask;
+            if (command == CommandWord.TODO) {
+                newTask = new Todo(contents);
+            } else if (command == CommandWord.DEADLINE) {
+                newTask = new Deadline(contents);
+            } else {
+                newTask = new Event(contents);
+            }
+            taskList.add(newTask);
+            System.out.println("Okay!\n" + "Added: " + newTask + "\n");
+        }
+
+        /**
+        * Handles the deletion of the task from the task list.
+        * @param inputWordsSplit String array of the command word with the task number
+        */
+        private static void deleteTask(String[] inputWordsSplit) {
+            try {
+                int taskNumber = Integer.parseInt(inputWordsSplit[1]);
+                Task task = taskList.get(taskNumber - 1);
+                System.out.println("Ok! I've removed this task: \n" + task + "\n");
+                taskList.remove(taskNumber - 1);
+            } catch (IndexOutOfBoundsException e) {
+                System.out.println("Please indicate a valid task number! :)\n");
+            }
+        }
+
+        /**
+         * Enum for the command word
+         */
+        private enum CommandWord {
+            LIST("list"),
+            MARK("mark"),
+            UNMARK("unmark"),
+            TODO("todo"),
+            DEADLINE("deadline"),
+            EVENT("event"),
+            DELETE("delete"),
+            BYE("bye");
+
+            // String field for comparing to the input to the enum
+            private final String command;
+
+            /**
+             * Constructor for the CommandWord enum
+             * @param command User input command
+             */
+            CommandWord(String command) {
+                this.command = command;
+            }
+
+            /**
+             * Compares the input command to the enum.
+             * @param input Input command string
+             * @return CommandWord from the enum that matches the input command
+             * @throws IllegalArgumentException When an invalid command is inputted
+             */
+            private static CommandWord getCommand(String input) throws IllegalArgumentException {
+                for (int i = 0; i < CommandWord.values().length; i++) {
+                    CommandWord temp = CommandWord.values()[i];
+                    if (input.equals(temp.command)) {
+                        return temp;
+                    }
+                }
+                throw new IllegalArgumentException("There is no such command called " + input + "!\n" + "Please try again :)\n");
```

Comment:
instead of throwing an `IllegalArgumentExcrption`, perhaps you can wrap the exception into a new `DukeException`.